### ISSUE TITLE
Go To Definition going to wrong file (which shouldn't be included)

### ISSUE THREAD

**Issue by montyevans** [microsoft/pylance-release#2219]
### VS Code version

1.63.2

### Extension version

v2021.12.1559732655

### OS type

macOS

### OS version

12.1

### Python distribution

Anaconda

### Python version

3.8.12

### Language server

Pylance

### Expected behaviour

Attempting to Go to Definition of a module that is not accessible to the current script should fail, and should be linted with a warning that the import could not be resolved.

### Actual behaviour

Attempting to Go to Definition of a module that is not accessible to the current script navigates to a module with the same name in a different directory, that is definitely not accessible by the current script. The issue persists even after restarting the Python Language Server.

The issue seems similar to one with the C++ extension, covered here: https://github.com/microsoft/vscode-cpptools/issues/2601

### Steps to reproduce

![test](https://user-images.githubusercontent.com/32969601/148136436-58a26734-82ca-405c-9538-4656709e0356.gif)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by heejaechang** [comment]
not sure how this can happen. we probably need log to find out as specified here - https://github.com/microsoft/pylance-release/blob/main/TROUBLESHOOTING.md#filing-an-issue

**Comment by erictraut** [comment]
Based on the screen capture, pylance is resolving the import, and "go to definition" is taking you to the resolved location. So it looks like it's working correctly.

If you need help understanding why pylance has resolved the import, we'll need additional details as @heejaechang suggested.

**Comment by bschnurr** [comment]
I think the issue is his workspace folder is "tmp" and its somehow searching outside of tmp into a sibling folder.

desktop
     -> tmp/main.py (workspace)
     -> datafolder/data.py

Do you have extra paths set?   Logs would help.

**Comment by montyevans** [comment]
Hey all. Thanks for taking a look at this, and sorry for the slow reply. 

`"python.analysis.extraPaths"` is currently unset, so I don't think Go to Definition should be searching outside the current folder. 

I just set `"python.analysis.logLevel": "Trace"`, and checked the Python Language Server section of the Output tab, to find:
```
Search paths for /Users/montyevans/Desktop/TMP
  /Users/montyevans/.vscode/extensions/ms-python.vscode-pylance-2022.1.0/dist/typeshed-fallback/stdlib
  /Users/montyevans/Desktop/TMP
  /Users/montyevans/Desktop/TMP/typings
  /Users/montyevans/.vscode/extensions/ms-python.vscode-pylance-2022.1.0/dist/typeshed-fallback/stubs/...
  /Users/montyevans/.vscode/extensions/ms-python.vscode-pylance-2022.1.0/dist/bundled/stubs
  /usr/local/Cellar/python@3.9/3.9.9/Frameworks/Python.framework/Versions/3.9/lib/python3.9
  /usr/local/Cellar/python@3.9/3.9.9/Frameworks/Python.framework/Versions/3.9/lib/python3.9/lib-dynload
  /Users/montyevans/Library/Python/3.9/lib/python/site-packages
  /usr/local/lib/python3.9/site-packages
  /Users/montyevans/Desktop/RaffelLab/rigl-reproducibility
  /usr/local/Cellar/sip/6.5.0/libexec/lib/python3.9/site-packages
```

It seems like `/Users/montyevans/Desktop/RaffelLab/rigl-reproducibility` is erroneously being added to search paths? I have been doing work (and using Go to Definition) within that folder recently, but I don't think that should cause Pylance to add it as an extra path.

I restarted VSCode, opened the folder, and then did Go to Definition on `data` as in the .gif. Full logs from that below:

---------

```
[Info  - 5:18:20 PM] Pylance language server 2022.1.0 (pyright 5829dcbb) starting
[Info  - 5:18:20 PM] Server root directory: /Users/montyevans/.vscode/extensions/ms-python.vscode-pylance-2022.1.0/dist
[Info  - 5:18:20 PM] No configuration file found.
[Info  - 5:18:20 PM] No pyproject.toml file found.
[Info  - 5:18:20 PM] Setting pythonPath for service "TMP": "/usr/local/bin/python3"
[Warn  - 5:18:20 PM] stubPath /Users/montyevans/Desktop/TMP/typings is not a valid directory.
[Info  - 5:18:20 PM] Assuming Python version 3.9
[Info  - 5:18:20 PM] Assuming Python platform Darwin
Search paths for /Users/montyevans/Desktop/TMP
  /Users/montyevans/.vscode/extensions/ms-python.vscode-pylance-2022.1.0/dist/typeshed-fallback/stdlib
  /Users/montyevans/Desktop/TMP
  /Users/montyevans/Desktop/TMP/typings
  /Users/montyevans/.vscode/extensions/ms-python.vscode-pylance-2022.1.0/dist/typeshed-fallback/stubs/...
  /Users/montyevans/.vscode/extensions/ms-python.vscode-pylance-2022.1.0/dist/bundled/stubs
  /usr/local/Cellar/python@3.9/3.9.9/Frameworks/Python.framework/Versions/3.9/lib/python3.9
  /usr/local/Cellar/python@3.9/3.9.9/Frameworks/Python.framework/Versions/3.9/lib/python3.9/lib-dynload
  /Users/montyevans/Library/Python/3.9/lib/python/site-packages
  /usr/local/lib/python3.9/site-packages
  /Users/montyevans/Desktop/RaffelLab/rigl-reproducibility
  /usr/local/Cellar/sip/6.5.0/libexec/lib/python3.9/site-packages
[Info  - 5:18:21 PM] Searching for source files
[Info  - 5:18:21 PM] Found 8 source files
[FG] parsing: /Users/montyevans/Desktop/TMP/main.py (17ms)
[FG] parsing: /Users/montyevans/.vscode/extensions/ms-python.vscode-pylance-2022.1.0/dist/typeshed-fallback/stdlib/builtins.pyi [fs read 2ms] (96ms)
[FG] binding: /Users/montyevans/.vscode/extensions/ms-python.vscode-pylance-2022.1.0/dist/typeshed-fallback/stdlib/builtins.pyi (36ms)
[FG] binding: /Users/montyevans/Desktop/TMP/main.py (2ms)
[Info  - 5:18:21 PM] Background analysis(1) root directory: /Users/montyevans/.vscode/extensions/ms-python.vscode-pylance-2022.1.0/dist
[Info  - 5:18:21 PM] Background analysis(1) started
Background analysis message: setConfigOptions
Background analysis message: setImportResolver
Background analysis message: ensurePartialStubPackages
Background analysis message: setTrackedFiles
Background analysis message: markAllFilesDirty
Background analysis message: setFileOpened
Background analysis message: getSemanticTokens full
[BG(1)] getSemanticTokens full at /Users/montyevans/Desktop/TMP/main.py ...
[BG(1)]   parsing: /Users/montyevans/Desktop/TMP/main.py (18ms)
[BG(1)]   parsing: /Users/montyevans/.vscode/extensions/ms-python.vscode-pylance-2022.1.0/dist/typeshed-fallback/stdlib/builtins.pyi [fs read 4ms] (84ms)
[BG(1)]   binding: /Users/montyevans/.vscode/extensions/ms-python.vscode-pylance-2022.1.0/dist/typeshed-fallback/stdlib/builtins.pyi (30ms)
[BG(1)]   binding: /Users/montyevans/Desktop/TMP/main.py (2ms)
[BG(1)]   parsing: /Users/montyevans/.vscode/extensions/ms-python.vscode-pylance-2022.1.0/dist/typeshed-fallback/stdlib/typing_extensions.pyi [fs read 0ms] (4ms)
[BG(1)]   binding: /Users/montyevans/.vscode/extensions/ms-python.vscode-pylance-2022.1.0/dist/typeshed-fallback/stdlib/typing_extensions.pyi (1ms)
[BG(1)]   parsing: /Users/montyevans/.vscode/extensions/ms-python.vscode-pylance-2022.1.0/dist/typeshed-fallback/stdlib/typing.pyi [fs read 0ms] (23ms)
[BG(1)]   binding: /Users/montyevans/.vscode/extensions/ms-python.vscode-pylance-2022.1.0/dist/typeshed-fallback/stdlib/typing.pyi (9ms)
[BG(1)]   parsing: /Users/montyevans/.vscode/extensions/ms-python.vscode-pylance-2022.1.0/dist/typeshed-fallback/stdlib/_typeshed/__init__.pyi [fs read 1ms] (8ms)
[BG(1)]   binding: /Users/montyevans/.vscode/extensions/ms-python.vscode-pylance-2022.1.0/dist/typeshed-fallback/stdlib/_typeshed/__init__.pyi (1ms)
[BG(1)]   parsing: /Users/montyevans/.vscode/extensions/ms-python.vscode-pylance-2022.1.0/dist/typeshed-fallback/stdlib/abc.pyi [fs read 0ms] (1ms)
[BG(1)]   binding: /Users/montyevans/.vscode/extensions/ms-python.vscode-pylance-2022.1.0/dist/typeshed-fallback/stdlib/abc.pyi (0ms)
[BG(1)]   parsing: /Users/montyevans/Desktop/RaffelLab/rigl-reproducibility/data.py [fs read 0ms] (22ms)
[BG(1)]   binding: /Users/montyevans/Desktop/RaffelLab/rigl-reproducibility/data.py ...
[BG(1)]     parsing: /Users/montyevans/Desktop/RaffelLab/rigl-reproducibility/sparselearning/utils/typing_alias.py [fs read 0ms] (13ms)
[BG(1)]     binding: /Users/montyevans/Desktop/RaffelLab/rigl-reproducibility/sparselearning/utils/typing_alias.py (0ms)
[BG(1)]   binding: /Users/montyevans/Desktop/RaffelLab/rigl-reproducibility/data.py (16ms)
[BG(1)] getSemanticTokens full at /Users/montyevans/Desktop/TMP/main.py (243ms)
Background analysis message: getSemanticTokens range
[BG(1)] getSemanticTokens range 0:0 - 4:0 at /Users/montyevans/Desktop/TMP/main.py (1ms)
Background analysis message: analyze
[BG(1)] analyzing: /Users/montyevans/Desktop/TMP/main.py ...
[BG(1)]   checking: /Users/montyevans/Desktop/TMP/main.py (5ms)
[BG(1)] analyzing: /Users/montyevans/Desktop/TMP/main.py (5ms)
Background analysis message: resumeAnalysis
[FG] parsing: /Users/montyevans/.vscode/extensions/ms-python.vscode-pylance-2022.1.0/dist/typeshed-fallback/stdlib/typing_extensions.pyi [fs read 1ms] (6ms)
[FG] binding: /Users/montyevans/.vscode/extensions/ms-python.vscode-pylance-2022.1.0/dist/typeshed-fallback/stdlib/typing_extensions.pyi (1ms)
[FG] parsing: /Users/montyevans/.vscode/extensions/ms-python.vscode-pylance-2022.1.0/dist/typeshed-fallback/stdlib/typing.pyi [fs read 1ms] (33ms)
[FG] binding: /Users/montyevans/.vscode/extensions/ms-python.vscode-pylance-2022.1.0/dist/typeshed-fallback/stdlib/typing.pyi (11ms)
[FG] parsing: /Users/montyevans/.vscode/extensions/ms-python.vscode-pylance-2022.1.0/dist/typeshed-fallback/stdlib/_typeshed/__init__.pyi [fs read 1ms] (10ms)
[FG] binding: /Users/montyevans/.vscode/extensions/ms-python.vscode-pylance-2022.1.0/dist/typeshed-fallback/stdlib/_typeshed/__init__.pyi (1ms)
[FG] parsing: /Users/montyevans/.vscode/extensions/ms-python.vscode-pylance-2022.1.0/dist/typeshed-fallback/stdlib/abc.pyi [fs read 0ms] (1ms)
[FG] binding: /Users/montyevans/.vscode/extensions/ms-python.vscode-pylance-2022.1.0/dist/typeshed-fallback/stdlib/abc.pyi (0ms)
Background analysis message: setFileOpened
Background analysis message: setFileClosed
Background analysis message: analyze
Background analysis message: setFileOpened
Background analysis message: setFileClosed
Background analysis message: setFileOpened
Background analysis message: setFileClosed
Background analysis message: analyze
Background analysis message: setFileOpened
[FG] parsing: /Users/montyevans/Desktop/RaffelLab/rigl-reproducibility/data.py (41ms)
[FG] binding: /Users/montyevans/Desktop/RaffelLab/rigl-reproducibility/data.py ...
[FG]   parsing: /Users/montyevans/Desktop/RaffelLab/rigl-reproducibility/sparselearning/utils/typing_alias.py [fs read 0ms] (19ms)
[FG]   binding: /Users/montyevans/Desktop/RaffelLab/rigl-reproducibility/sparselearning/utils/typing_alias.py (0ms)
[FG] binding: /Users/montyevans/Desktop/RaffelLab/rigl-reproducibility/data.py (24ms)
Background analysis message: getDiagnosticsForRange
Background analysis message: getDiagnosticsForRange
Background analysis message: getSemanticTokens full
[BG(1)] getSemanticTokens full at /Users/montyevans/Desktop/RaffelLab/rigl-reproducibility/data.py ...
[BG(1)]   parsing: /Users/montyevans/.vscode/extensions/ms-python.vscode-pylance-2022.1.0/dist/typeshed-fallback/stdlib/logging/__init__.pyi [fs read 1ms] (49ms)
[BG(1)]   binding: /Users/montyevans/.vscode/extensions/ms-python.vscode-pylance-2022.1.0/dist/typeshed-fallback/stdlib/logging/__init__.pyi (6ms)
[BG(1)]   parsing: /Users/montyevans/.vscode/extensions/ms-python.vscode-pylance-2022.1.0/dist/typeshed-fallback/stdlib/dataclasses.pyi [fs read 0ms] (6ms)
[BG(1)]   binding: /Users/montyevans/.vscode/extensions/ms-python.vscode-pylance-2022.1.0/dist/typeshed-fallback/stdlib/dataclasses.pyi (1ms)
[BG(1)]   parsing: /Users/montyevans/.vscode/extensions/ms-python.vscode-pylance-2022.1.0/dist/typeshed-fallback/stdlib/math.pyi [fs read 0ms] (4ms)
[BG(1)]   binding: /Users/montyevans/.vscode/extensions/ms-python.vscode-pylance-2022.1.0/dist/typeshed-fallback/stdlib/math.pyi (2ms)
[BG(1)]   parsing: /Users/montyevans/.vscode/extensions/ms-python.vscode-pylance-2022.1.0/dist/typeshed-fallback/stdlib/pathlib.pyi [fs read 0ms] (6ms)
[BG(1)]   binding: /Users/montyevans/.vscode/extensions/ms-python.vscode-pylance-2022.1.0/dist/typeshed-fallback/stdlib/pathlib.pyi (4ms)
[FG] parsing: /Users/montyevans/.vscode/extensions/ms-python.vscode-pylance-2022.1.0/dist/typeshed-fallback/stdlib/math.pyi [fs read 0ms] (9ms)
[FG] binding: /Users/montyevans/.vscode/extensions/ms-python.vscode-pylance-2022.1.0/dist/typeshed-fallback/stdlib/math.pyi (2ms)
[BG(1)]   parsing: /Users/montyevans/.vscode/extensions/ms-python.vscode-pylance-2022.1.0/dist/typeshed-fallback/stdlib/os/__init__.pyi [fs read 1ms] (48ms)
[BG(1)]   binding: /Users/montyevans/.vscode/extensions/ms-python.vscode-pylance-2022.1.0/dist/typeshed-fallback/stdlib/os/__init__.pyi (15ms)
[BG(1)]   parsing: /usr/local/lib/python3.9/site-packages/numpy/__init__.pyi [fs read 1ms] (148ms)
[BG(1)]   binding: /usr/local/lib/python3.9/site-packages/numpy/__init__.pyi (34ms)
[BG(1)]   parsing: /usr/local/lib/python3.9/site-packages/torch/utils/data/__init__.py [fs read 0ms] (7ms)
[BG(1)]   binding: /usr/local/lib/python3.9/site-packages/torch/utils/data/__init__.py (2ms)
[BG(1)]   parsing: /usr/local/lib/python3.9/site-packages/torch/utils/data/dataset.py [fs read 0ms] (14ms)
[BG(1)]   binding: /usr/local/lib/python3.9/site-packages/torch/utils/data/dataset.py (5ms)
[BG(1)]   parsing: /usr/local/lib/python3.9/site-packages/torch/utils/data/dataloader.py [fs read 0ms] (28ms)
[BG(1)]   binding: /usr/local/lib/python3.9/site-packages/torch/utils/data/dataloader.py (6ms)
[BG(1)]   parsing: /usr/local/lib/python3.9/site-packages/torchvision/__init__.py [fs read 0ms] (4ms)
[BG(1)]   binding: /usr/local/lib/python3.9/site-packages/torchvision/__init__.py (1ms)
[BG(1)]   parsing: /usr/local/lib/python3.9/site-packages/numpy/core/multiarray.pyi [fs read 0ms] (18ms)
[BG(1)]   binding: /usr/local/lib/python3.9/site-packages/numpy/core/multiarray.pyi (5ms)
[BG(1)]   parsing: /usr/local/lib/python3.9/site-packages/numpy/typing/__init__.py [fs read 1ms] (5ms)
[BG(1)]   binding: /usr/local/lib/python3.9/site-packages/numpy/typing/__init__.py (2ms)
[BG(1)]   parsing: /usr/local/lib/python3.9/site-packages/numpy/typing/_dtype_like.py [fs read 0ms] (4ms)
[BG(1)]   binding: /usr/local/lib/python3.9/site-packages/numpy/typing/_dtype_like.py (2ms)
[BG(1)]   parsing: /usr/local/lib/python3.9/site-packages/numpy/typing/_generic_alias.py [fs read 0ms] (7ms)
[BG(1)]   binding: /usr/local/lib/python3.9/site-packages/numpy/typing/_generic_alias.py (3ms)
[BG(1)]   parsing: /usr/local/lib/python3.9/site-packages/numpy/typing/_shape.py [fs read 0ms] (1ms)
[BG(1)]   binding: /usr/local/lib/python3.9/site-packages/numpy/typing/_shape.py (0ms)
[BG(1)]   parsing: /Users/montyevans/.vscode/extensions/ms-python.vscode-pylance-2022.1.0/dist/typeshed-fallback/stdlib/enum.pyi [fs read 0ms] (4ms)
[BG(1)]   binding: /Users/montyevans/.vscode/extensions/ms-python.vscode-pylance-2022.1.0/dist/typeshed-fallback/stdlib/enum.pyi (2ms)
[BG(1)]   parsing: /usr/local/lib/python3.9/site-packages/numpy/typing/_array_like.py [fs read 0ms] (1ms)
[BG(1)]   binding: /usr/local/lib/python3.9/site-packages/numpy/typing/_array_like.py (1ms)
[BG(1)]   parsing: /usr/local/lib/python3.9/site-packages/numpy/typing/_nested_sequence.py [fs read 1ms] (4ms)
[BG(1)]   binding: /usr/local/lib/python3.9/site-packages/numpy/typing/_nested_sequence.py (0ms)
[BG(1)]   parsing: /usr/local/lib/python3.9/site-packages/torchvision/transforms/__init__.py [fs read 0ms] (0ms)
[BG(1)]   binding: /usr/local/lib/python3.9/site-packages/torchvision/transforms/__init__.py ...
[BG(1)]     parsing: /usr/local/lib/python3.9/site-packages/torchvision/transforms/transforms.py [fs read 1ms] (31ms)
[BG(1)]     binding: /usr/local/lib/python3.9/site-packages/torchvision/transforms/transforms.py (17ms)
[BG(1)]     parsing: /usr/local/lib/python3.9/site-packages/torchvision/transforms/autoaugment.py [fs read 0ms] (44ms)
[BG(1)]     binding: /usr/local/lib/python3.9/site-packages/torchvision/transforms/autoaugment.py (4ms)
[BG(1)]   binding: /usr/local/lib/python3.9/site-packages/torchvision/transforms/__init__.py (96ms)
[BG(1)]   parsing: /usr/local/lib/python3.9/site-packages/torch/__init__.py [fs read 0ms] (31ms)
[BG(1)]   binding: /usr/local/lib/python3.9/site-packages/torch/__init__.py ...
[BG(1)]     parsing: /usr/local/lib/python3.9/site-packages/torch/_C/__init__.pyi [fs read 1ms] (39ms)
[BG(1)]     binding: /usr/local/lib/python3.9/site-packages/torch/_C/__init__.pyi (31ms)
[BG(1)]     parsing: /usr/local/lib/python3.9/site-packages/torch/_C/_VariableFunctions.pyi [fs read 1ms] (38ms)
[BG(1)]     binding: /usr/local/lib/python3.9/site-packages/torch/_C/_VariableFunctions.pyi (18ms)
[BG(1)]     parsing: /usr/local/lib/python3.9/site-packages/torch/functional.py [fs read 0ms] (12ms)
[BG(1)]     binding: /usr/local/lib/python3.9/site-packages/torch/functional.py (5ms)
[BG(1)]   binding: /usr/local/lib/python3.9/site-packages/torch/__init__.py (148ms)
[BG(1)]   parsing: /usr/local/lib/python3.9/site-packages/torch/nn/__init__.py [fs read 0ms] (1ms)
[BG(1)]   binding: /usr/local/lib/python3.9/site-packages/torch/nn/__init__.py ...
[BG(1)]     parsing: /usr/local/lib/python3.9/site-packages/torch/nn/modules/__init__.py [fs read 1ms] (2ms)
[BG(1)]     binding: /usr/local/lib/python3.9/site-packages/torch/nn/modules/__init__.py (1ms)
[BG(1)]   binding: /usr/local/lib/python3.9/site-packages/torch/nn/__init__.py (3ms)
[BG(1)]   parsing: /usr/local/lib/python3.9/site-packages/torch/nn/modules/module.py [fs read 1ms] (11ms)
[BG(1)]   binding: /usr/local/lib/python3.9/site-packages/torch/nn/modules/module.py (9ms)
[BG(1)]   parsing: /usr/local/lib/python3.9/site-packages/torchvision/datasets/__init__.py [fs read 1ms] (3ms)
[BG(1)]   binding: /usr/local/lib/python3.9/site-packages/torchvision/datasets/__init__.py (0ms)
[BG(1)]   parsing: /usr/local/lib/python3.9/site-packages/torchvision/datasets/cifar.py [fs read 0ms] (7ms)
[BG(1)]   binding: /usr/local/lib/python3.9/site-packages/torchvision/datasets/cifar.py (1ms)
[BG(1)]   parsing: /usr/local/lib/python3.9/site-packages/torchvision/datasets/vision.py [fs read 0ms] (2ms)
[BG(1)]   binding: /usr/local/lib/python3.9/site-packages/torchvision/datasets/vision.py (1ms)
[BG(1)]   parsing: /usr/local/lib/python3.9/site-packages/torchvision/transforms/functional.py [fs read 1ms] (22ms)
[BG(1)]   binding: /usr/local/lib/python3.9/site-packages/torchvision/transforms/functional.py (9ms)
[BG(1)]   parsing: /Users/montyevans/.vscode/extensions/ms-python.vscode-pylance-2022.1.0/dist/typeshed-fallback/stdlib/warnings.pyi [fs read 0ms] (4ms)
[BG(1)]   binding: /Users/montyevans/.vscode/extensions/ms-python.vscode-pylance-2022.1.0/dist/typeshed-fallback/stdlib/warnings.pyi (1ms)
[BG(1)]   parsing: /Users/montyevans/.vscode/extensions/ms-python.vscode-pylance-2022.1.0/dist/typeshed-fallback/stdlib/_warnings.pyi [fs read 0ms] (0ms)
[BG(1)]   binding: /Users/montyevans/.vscode/extensions/ms-python.vscode-pylance-2022.1.0/dist/typeshed-fallback/stdlib/_warnings.pyi (1ms)
[BG(1)]   parsing: /usr/local/lib/python3.9/site-packages/torchvision/datasets/folder.py [fs read 0ms] (5ms)
[BG(1)]   binding: /usr/local/lib/python3.9/site-packages/torchvision/datasets/folder.py (2ms)
[BG(1)]   parsing: /usr/local/lib/python3.9/site-packages/torchvision/datasets/mnist.py [fs read 0ms] (13ms)
[BG(1)]   binding: /usr/local/lib/python3.9/site-packages/torchvision/datasets/mnist.py (4ms)
[BG(1)]   parsing: /Users/montyevans/.vscode/extensions/ms-python.vscode-pylance-2022.1.0/dist/typeshed-fallback/stdlib/_collections_abc.pyi [fs read 1ms] (1ms)
[BG(1)]   binding: /Users/montyevans/.vscode/extensions/ms-python.vscode-pylance-2022.1.0/dist/typeshed-fallback/stdlib/_collections_abc.pyi (1ms)
[BG(1)]   parsing: /Users/montyevans/.vscode/extensions/ms-python.vscode-pylance-2022.1.0/dist/typeshed-fallback/stdlib/types.pyi [fs read 0ms] (5ms)
[BG(1)]   binding: /Users/montyevans/.vscode/extensions/ms-python.vscode-pylance-2022.1.0/dist/typeshed-fallback/stdlib/types.pyi (2ms)
[BG(1)]   parsing: /Users/montyevans/.vscode/extensions/ms-python.vscode-pylance-2022.1.0/dist/typeshed-fallback/stdlib/collections/abc.pyi [fs read 0ms] (0ms)
[BG(1)]   binding: /Users/montyevans/.vscode/extensions/ms-python.vscode-pylance-2022.1.0/dist/typeshed-fallback/stdlib/collections/abc.pyi (0ms)
[BG(1)]   parsing: /usr/local/lib/python3.9/site-packages/numpy/lib/npyio.pyi [fs read 0ms] (3ms)
[BG(1)]   binding: /usr/local/lib/python3.9/site-packages/numpy/lib/npyio.pyi (2ms)
[BG(1)]   parsing: /usr/local/lib/python3.9/site-packages/numpy/random/__init__.pyi [fs read 0ms] (9ms)
[BG(1)]   binding: /usr/local/lib/python3.9/site-packages/numpy/random/__init__.pyi (1ms)
[BG(1)]   parsing: /usr/local/lib/python3.9/site-packages/numpy/random/mtrand.pyi [fs read 0ms] (4ms)
[BG(1)]   binding: /usr/local/lib/python3.9/site-packages/numpy/random/mtrand.pyi (7ms)
[BG(1)]   parsing: /usr/local/lib/python3.9/site-packages/torch/utils/data/sampler.py [fs read 0ms] (1ms)
[BG(1)]   binding: /usr/local/lib/python3.9/site-packages/torch/utils/data/sampler.py (2ms)
[BG(1)] getSemanticTokens full at /Users/montyevans/Desktop/RaffelLab/rigl-reproducibility/data.py (1135ms)
Background analysis message: getSemanticTokens range
[BG(1)] getSemanticTokens range 0:0 - 45:0 at /Users/montyevans/Desktop/RaffelLab/rigl-reproducibility/data.py (6ms)
Background analysis message: setFileOpened
Background analysis message: setFileClosed
Background analysis message: analyze
[BG(1)] analyzing: /Users/montyevans/Desktop/RaffelLab/rigl-reproducibility/data.py ...
[BG(1)]   checking: /Users/montyevans/Desktop/RaffelLab/rigl-reproducibility/data.py (30ms)
[BG(1)] analyzing: /Users/montyevans/Desktop/RaffelLab/rigl-reproducibility/data.py (30ms)
Background analysis message: resumeAnalysis
```

**Comment by erictraut** [comment]
You must have configured your Python environment to include that as an import resolution path.

If you run python from your shell, you can see what paths it is using.
1. From the shell, activate the Python virtual environment you're using (assuming you're using a venv)
2. Start the python interpreter: "python3"
3. "import sys"
4. "print(sys.path)"

Perhaps you did an editable install of a package in that directory? Or do you have a PYTHON_PATH environment variable declared?

**Comment by montyevans** [comment]
Aha, you're completely right - I did an editable install a while ago that added `RaffelLab/rigl-reproducibility` to my local Python3's `sys.path`.

I was getting confused because my integrated terminal was running a conda environment (whose `sys.path` was unedited), but VSCode was still using the local Python version (which had the editable install affecting `sys.path`). So when I checked `sys.path` in the terminal, nothing seemed wrong... until I realized that I was looking at two different Pythons.

Clearly I need to be more conscientious about keeping my python versions in sync, but might it be an idea to head off this sort of error by providing some kind of warning when VSCode is running a different python version that the integrated terminal? Or even having `conda activate myenv` cause VSCode to automatically change to the corresponding interpreter?

Thanks again for walking me through this.
